x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupOperator.java

```diff
@@ -107,9 +119,117 @@ protected void performAsync(Page inputPage, ActionListener<Page> listener) {
         );
     }
 
+    @Override
+    public String toString() {
+        return "EnrichOperator[index="
+            + enrichIndex
+            + " match_field="
+            + matchField
+            + " enrich_fields="
+            + enrichFields
+            + " inputChannel="
+            + inputChannel
+            + "]";
+    }
+
     @Override
     protected void doClose() {
         // TODO: Maybe create a sub-task as the parent task of all the lookup tasks
         // then cancel it when this operator terminates early (e.g., have enough result).
     }
+
+    @Override
+    public Operator.Status status() {
+        final PageStats stats = pageStats();
+        assert stats.received() >= stats.completed() : stats.received() + " < " + stats.completed();
+        return new Status(stats.received(), stats.completed(), totalTerms, TimeValue.timeValueNanos(totalTimeInNanos()).millis());
+    }
```

Comment:
In the other operators I did this by making method that receives all of the parts of the stats on the superclass and letting subclasses override it. So they have to think about what they want to do with all the bits rather than look around for the bits they want. I do think that's a better way to do it but I'm ok either eway.

Reply:
++, good suggestion.